tests/test_quantile_encoder.py

```diff
@@ -65,6 +61,58 @@ def test_new_category(self):
 
         new_medians = pd.DataFrame({"categories": [3.0, 3.0]})
 
-        pd.testing.assert_frame_equal(
-            transformer_median.transform(new_df), new_medians
+        pd.testing.assert_frame_equal(transformer_median.transform(new_df), new_medians)
+
+
+class TestSummaryEncoder(unittest.TestCase):
+    """Tests for percentile encoder."""
+
+    def setUp(self):
+        """Create dataframe with categories and a target variable"""
+
+        self.df = pd.DataFrame({"categories": ["a", "b", "c", "a", "b", "c", "a", "b"]})
+        self.target = np.array([1, 2, 0, 4, 5, 0, 6, 7])
+        self.col = "categories"
+
+    def assert_same_quantile(self, quantile):
+
+        quantile_results = encoders.QuantileEncoder(
+            cols=[self.col], quantile=quantile
+        ).fit_transform(self.df, self.target)
+
+        summary_results = encoders.SummaryEncoder(
+            cols=[self.col], quantiles=[quantile]
+        ).fit_transform(self.df, self.target)
+
+        percentile = round(quantile * 100)
+
+        np.testing.assert_allclose(
+            quantile_results[self.col].values,
+            summary_results[f"{self.col}_{percentile}"].values,
         )
+
+    def test_several_quantiles(self):
+
+        for quantile in [0.1, 0.5, 0.9]:
+            self.assert_same_quantile(quantile)
+
+    def test_several_quantiles(self):
```

Comment:
this test has the same name as the above one. Is the above one really necessary? If so we should probably rename them